state_synchronizer/src/peer_manager.rs

```diff
@@ -94,28 +101,51 @@ impl PeerManager {
         if let Some(peer_info) = self.peers.get_mut(peer_id) {
             match update_type {
                 PeerScoreUpdateType::Success => {
-                    let new_score = peer_info.score + 1.0;
-                    peer_info.score = new_score.min(MAX_SCORE);
+                    if peer_info.score < MAX_SCORE {
+                        let new_score = peer_info.score + 1.0;
+                        peer_info.score = new_score.min(MAX_SCORE);
+                        self.weighted_index = self.compute_weighted_index();
```

Comment:
do this outside the match?

Reply:
yeah, you can just get old score. Perform update and compare it